__tests__/app.test.js

```diff
@@ -4,51 +4,255 @@ const request = require("supertest");
+        return request(app)
+            .get(`/api/articles/${articleId}/comments`)
+            .expect(200)
+            .then((res) => {
+                const { body } = res;
+                const { comments } = body;
+                expect(Array.isArray(comments)).toBe(true);
+                expect(comments.length).toBe(0);
+            });
+    });
+});
+
+describe('POST /api/articles/:article_id/comments', () => {
+    it('responds with the posted comment', () => {
+        const articleId = 1;
+        const newComment = {
+            username: data.userData[0].username,
+            body: 'This is a test comment.'
+        };
+        return request(app)
+            .post(`/api/articles/${articleId}/comments`)
+            .send(newComment)
+            .expect(201)
+            .then((res) => {
+                const { body } = res;
+                expect(body).toHaveProperty('comment');
+                const { comment } = body;
+                expect(comment).toEqual(expect.objectContaining({
+                    comment_id: expect.any(Number),
+                    author: newComment.username,
+                    body: newComment.body,
+                    article_id: articleId,
+                    created_at: expect.any(String)
+                }))
+
+            })
+    });
+
+    it('responds with status 404 if article_id is not found', () => {
+        const articleId = 999999;
+        const newComment = {
+            username: 'testuser',
+            body: 'This is a test comment.'
+        };
+        return request(app)
+            .post(`/api/articles/${articleId}/comments`)
+            .send(newComment)
+            .expect(404)
+            .then((res) => {
+                const { body } = res;
+                expect(body.msg).toBe(`Article ${articleId} not found`);
+            });
+    });
+
+    it('responds with status 400 if article_id is not a number', () => {
+        const articleId = 'not_a_number';
+        const newComment = {
+            username: 'testuser',
+            body: 'This is a test comment.'
+        };
+        return request(app)
+            .post(`/api/articles/${articleId}/comments`)
+            .send(newComment)
+            .expect(400)
+            .then((res) => {
+                const { body } = res;
+                const { msg } = body;
+                expect(msg).toBe('Invalid article ID');
+            });
+    });
+
+    it('responds with status 400 if request body is missing required properties', () => {
+        const articleId = 1;
+        const invalidComment = {
+            username: 'testuser'
+        };
+        return request(app)
+            .post(`/api/articles/${articleId}/comments`)
+            .send(invalidComment)
+            .expect(400)
+            .then((res) => {
+                const { body } = res;
+                const { msg } = body;
+                expect(msg).toBe('Invalid request body');
+            });
     });
```

Comment:
two tests to add: 
what if the sent comment has a property that is unnecessary? (201 or 400)
what if the username entered does not exist?

Reply:
ive added two more tests